internal/core/metadata/rest_command.go

```diff
@@ -82,3 +82,28 @@ func restGetCommandByName(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(results)
 }
+
+func restGetCommandsByDeviceId(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	did, err := url.QueryUnescape(vars[ID])
+	if err != nil {
+		LoggingClient.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	commands, err := dbClient.GetCommandsByDeviceId(did)
+	if err != nil {
+		if err == db.ErrNotFound {
+			http.Error(w, err.Error(), http.StatusNotFound)
+		} else {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+
+		LoggingClient.Error(err.Error())
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(commands)
+
```

Comment:
Given that we are in the process of extending unit test coverage through the use of what we're calling the "operator" pattern, I'd like to see this new endpoint refactored according to that pattern. Otherwise it will be tech debt we have to go back and address. The check for `db.ErrNotFound` should be inside the operator and then the router layer should perform type checking to determine the appropriate response code. This example should illustrate what I'm looking for.
https://github.com/edgexfoundry/edgex-go/blob/a478f95df2952767d2a1b82a35f69e280022e99c/internal/core/metadata/rest_device.go#L38

Then, to really finish it off, add a test for the router logic as well. Example:
https://github.com/edgexfoundry/edgex-go/blob/a478f95df2952767d2a1b82a35f69e280022e99c/internal/core/metadata/rest_device_test.go#L30

Reply:
Ok, I will rewrite it and ... will add the missing RAML